### ISSUE TITLE
Calling a trait function without `impl` causes an panick

### ISSUE THREAD

**Issue by pt9999** [tttmmmyyyy/fixlang#43]
Calling a trait function without `impl` causes an panick.
```
thread 'main' panicked at 'called `Option::unwrap()` on a `None` value', src/typecheck.rs:1147:70
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

Sample program:
```
module Main;

trait a: Foo {
    foo: a -> IO ();
}

main: IO ();
main = (
    123.foo
);

```

**Comment by tttmmmyyyy** [comment]
Fixed at f1eb9376a3d22f79847328f24a7543d0dba7ca2e.

**Comment by tttmmmyyyy** [comment]
Fixed at f1eb9376a3d22f79847328f24a7543d0dba7ca2e.
